mlir/lib/Dialect/MIOpen/Transforms/AffixTuningParameters.cpp

```diff
@@ -45,20 +52,19 @@ void AffixTuningParameters::runOnOperation() {
 
   func.walk([&](miopen::Conv2DOp op) { affixTuningParametersImpl(op); });
   func.walk([&](miopen::Conv2DBwdDataOp op) { affixTuningParametersImpl(op); });
-  func.walk(
-      [&](miopen::Conv2DBwdWeightOp op) { affixTuningParametersImpl(op); });
+  func.walk([&](miopen::Conv2DBwdWeightOp op) {
+    affixTuningParametersImpl(op);
+    affixBackwardWeightUtilityKernels(op);
+  });
 }
 
 template <typename T>
-void AffixTuningParameters::affixTuningParametersImpl(T &op) {
-  OpBuilder b(op.getContext());
-
-  ConvolutionContext convContext = populateConvContext(op);
-
+std::tuple<int64_t, int64_t, int64_t, int64_t, int64_t, int64_t, int64_t,
+           int64_t, int64_t>
```

Comment:
Just as a sanity check, do we perhaps want to make this some sort of `struct ConvDims {...}` instead? Or at least to document the order of the arguments in this tuple. int64_t^{big} just feels like a bad return type to me.

Reply:
I'd like to keep this as-is for now. In https://github.com/ROCmSoftwarePlatform/llvm-project-private/issues/439 we'd consolidate data structures commonly used between the generator and the lowering pipeline.

Reply:
Fair enough, and if it works, it works.